### PR TITLE
Introduce a signal handler for printing thread stack traces

### PR DESCRIPTION
**Pull request by vondele**
as a debugging aid, introduce a signal handler for printing stack traces

`kill -USR1 <pid>`

will result in a stack dump

needs testing on dev. The signal handler must be installed from the main python thread.

### CODE REVIEW

File: server/fishtest/__init__.py [hunk 1]
```diff
@@ -8,12 +8,23 @@
 from pyramid.config import Configurator
 from pyramid.events import BeforeRender, NewRequest
 from pyramid.session import SignedCookieSessionFactory
-
 from fishtest import helpers
+import traceback, signal, sys
+
+
+def thread_stack_dump(sig, frame):
+    for th in threading.enumerate():
```
Comment (ppigazzini) [inline server/fishtest/__init__.py:16]: Reading from mobile, if I'm not wrong we should import `threading`

**Comment by vondele** [comment]
updated to current master. Add a flush for better understandable output, and catch exceptions that might happen if threads do no longer exist. 

Good for merge IMO.

**Review by ppigazzini — APPROVED** [review]
Works fine on PROD

**Comment by ppigazzini** [comment]
PROD updated, thank you @vondele

**Pull request merged**